### ISSUE TITLE
Pagination ?

### ISSUE THREAD

**Issue by kanthi** [alex-shpak/hugo-book#165]
Is it possible to add pagination for entire book ?

**Comment by alex-shpak** [comment]
Hi!
Can you please elaborate? Blog posts are paginated, but docs sections are not rendered as lists rather as standalone pages.

**Comment by kanthi** [comment]
Hi
As you have mentioned I am talking about pagination in docs/book section. Its very difficult to navigate for large documentation in my exp. Also a layout change for blog (removing left side navigation ) would be great.  KISS is good, in my opinion blog layout at the moment is not ideal for someone who want to use it for both blog & docs/book.

**Comment by alex-shpak** [comment]
So, what kind of pagination you have in mind, next and previous page?
Having paginated list of all pages? Now there are no lists of docs pages to paginate.

As for menu for blog, left menu also holds search and site title, I think it's not wise to remove it. But you can customize `menu` template for type `posts` as you like it.

**Comment by okapixel** [comment]
Hi.
I also want the pagination feature. In particular, it would be nice to have the next/prev link in the book section as well.

**Comment by alex-shpak** [comment]
Some work have been done in `feature/prev-next` branch, but I didn't find yet suitable design and place for navigation buttons.

**Comment by kadamski** [comment]
@alex-shpak The implementation in feature/prev-next is almost exactly what I need (to be 100% happy, I would like to be able to disable it for some pages). The placement of the navigation buttons seems fine for me. What is stopping you from introducing this feature to master branch (could be experimental opt-in like the two existing experimental features).

**Comment by alex-shpak** [comment]
Hi!
Well, two points:
1. I don't like implementation, particularly [here](https://github.com/alex-shpak/hugo-book/blob/feature/prev-next/layouts/partials/docs/prev-next-cache.html), generating list of pages in memory. might be bad for big websites.
2. Placement/design of buttons. I think it requires more changes, perhaps language switch should be moved to the top of the page, otherwise it gets quite crowded in the footer.

**Comment by laoshaw** [comment]
Yes please add prev next navigation buttons and make left sidebar scrollable, e.g. https://v3.vuejs.org/guide/introduction.html  (which uses hexo)

**Comment by alex-shpak** [comment]
I will close issue for now, unfortunately I don't have scalable solution for it in hugo

**Comment by quite** [comment]
Ouch, so that feature/next-prev branch was deemed a dead end? @alex-shpak

**Comment by chinacomx** [comment]
dear @alex-shpak, were super happy with the theme for our translation page (https://chinacomx.github.io/translations/) and also were using the prev/next feature for a while, but now its gone again. any updates on the process? really really much apprecited

**Comment by cjauvin** [comment]
Me and @lemire are using this theme for our e-courses, and we need this feature, as Moodle has it already. I have made a fork with it: 

https://github.com/cjauvin/hugo-book-teluq

**Comment by lemire** [comment]
@alex-shpak 

> I don't like implementation, particularly [here](https://github.com/alex-shpak/hugo-book/blob/feature/prev-next/layouts/partials/docs/prev-next-cache.html), generating list of pages in memory. might be bad for big websites.

How much memory are we talking about ?

**Comment by whalekeykeeper** [comment]
> Me and [@lemire](https://github.com/lemire) are using this theme for our e-courses, and we need this feature, as Moodle has it already. I have made a fork with it:
> 
> https://github.com/cjauvin/hugo-book-teluq

huhu thanks a lot! Is it already working?

**Comment by cjauvin** [comment]
@whalekeykeeper Yes we are already using it, you can just download it as a submodule in your `themes` folder:

`$ git submodule add git@github.com:cjauvin/hugo-book-teluq.git ./themes/hugo-book-teluq`

and then set `theme = 'hugo-book-teluq'` in your config file. 

Just to be clear, this is just a fork of https://github.com/alex-shpak/hugo-book, along with the changes made in the branch https://github.com/alex-shpak/hugo-book/tree/feature/prev-next, and some minor fixes.

**Comment by whalekeykeeper** [comment]
> [@whalekeykeeper](https://github.com/whalekeykeeper) Yes we are already using it, you can just download it as a submodule in your `themes` folder:
> 
> `$ git submodule add git@github.com:cjauvin/hugo-book-teluq.git ./themes/hugo-book-teluq`
> 
> and then set `theme = 'hugo-book-teluq'` in your config file.
> 
> Just to be clear, this is just a fork of https://github.com/alex-shpak/hugo-book, along with the changes made in the branch https://github.com/alex-shpak/hugo-book/tree/feature/prev-next, and some minor fixes.

Will have a try. Thank you so much!

**Comment by lemire** [comment]
@whalekeykeeper You can see in action here... https://lemire.github.io/inf1220-hugo/ (In French but it should not matter to you.)

It works quite well.

**Comment by alex-shpak** [comment]
I see this issue get a lot of traction, I will re-open it

**Comment by alex-shpak** [comment]
`feature/prev-next` branch has also been rebased on latest `master`

**Comment by alex-shpak** [comment]
Well, it has been merged
